### PR TITLE
Feature/activity-backend-tests

### PR DESCRIPTION
**Pull request by ndeasy**
# Description

Added backend tests for Activity

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: activity-recorder/src/routes/__tests__/activity.test.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+        const cookie2 = global.signin();
+        await postActivity(cookie2);
+
+        expect(await Activity.find({})).toHaveLength(3);
+
+        const response = await request(app).get('/api/activity').set('Cookie', cookie1).send().expect(200);
+        expect(response.body).toHaveLength(2);
+    });
+});
```
Comment (hb308) [inline activity-recorder/src/routes/__tests__/activity.test.ts:89]: Really good work on the tests! Do we have tests for deleting an activity?

E.g. trying to delete an activity which doesn't exist, deleting an activity if not authorised, successful deletion etc

Comment (ndeasy) [reply activity-recorder/src/routes/__tests__/activity.test.ts:89]: Good point, missed that one. I'll add it

Comment (ndeasy) [reply activity-recorder/src/routes/__tests__/activity.test.ts:89]: Removed delete route as I couldn't get the mongoose id to work correctly...

Comment (hb308) [reply activity-recorder/src/routes/__tests__/activity.test.ts:89]: Have you tried to first create an activity in the test and then assign it to a const variable? From there you should be able to retrieve the activity id and then pass that as a parameter to the delete activity endpoint

**Review by hb308 — APPROVED** [review]
Delete route has been added with tests now :) Feel free to merge

**Pull request merged**